src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -2353,10 +2353,9 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           StarWithDefaultError(meth)
 
         if (!isPastTyper) {
-          val allParams = meth.paramss.flatten
-          for (p <- allParams) {
+          for (pp <- meth.paramss ; p <- pp){
             for (n <- p.deprecatedParamName) {
-              if (allParams.exists(p1 => p != p1 && (p1.name == n || p1.deprecatedParamName.exists(_ == n))))
```

Comment:
Looks a bit clearner with our [combinator for this](https://github.com/scala/scala/blob/ed41f9659ced08c2e1b64021ab22a3b04107a279/src/reflect/scala/reflect/internal/util/Collections.scala#L38-L39):

`mexists(p1 => p != p1 && (p1.name == n || p1.deprecatedParamName.exists(_ == n))`